extensions/amp-access/0.1/login-dialog.js

```diff
@@ -46,6 +46,21 @@ export function openLoginDialog(ampdoc, urlOrPromise) {
   return new WebLoginDialog(ampdoc.win, viewer, urlOrPromise).open();
 }
 
+/**
+ * Gets the final login URL with all the performed replacements.
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {string|!Promise<string>} urlOrPromise
+ * @return {!Promise<string>}
+ */
+export function getLoginUrl(ampdoc, urlOrPromise) {
+  const viewer = viewerForDoc(ampdoc);
```

Comment:
Could you please create an internal function and use it both here and in `openLoginDialog`. It'd look something like this:

```
/**
 * @param ...
 * @return {!WebLoginDialog|!ViewerLoginDialog}
 */
function createLoginDialog(ampdoc, urlOrPromise) {
  const viewer = viewerForDoc(ampdoc);
  const overrideDialog = parseInt(viewer.getParam('dialog'), 10);
  if (overrideDialog) {
    return new ViewerLoginDialog(viewer, urlOrPromise);
  }
  return new WebLoginDialog(ampdoc.win, viewer, urlOrPromise);
}

export function openLoginDialog(ampdoc, urlOrPromise) {
  return createLoginDialog(ampdoc, urlOrPromise).open();
}
export function getLoginUrl(ampdoc, urlOrPromise) {
  return createLoginDialog(ampdoc, urlOrPromise).getLoginUrl();
}
```